modules/core/shared/src/main/scala/eu/timepit/refined/numeric.scala

```diff
@@ -55,6 +58,18 @@ object numeric extends NumericValidate with NumericInference {
   /** Predicate that checks if a numeric value is zero or positive (>= 0). */
   type NonNegative = Not[Negative]
 
+  /** Predicate that checks if a numeric value is evenly divisible by `N`. */
+  type Divisible[N] = Modulo[N, _0]
+
+  /** Predicate that checks if a numeric value is not evenly divisible by `N`. */
+  type NonDivisible[N] = Not[Divisible[N]]
+
+  /** Predicate that checks if a numeric value is evenly divisible by 2. */
+  type Even = Divisible[_2]
+
+  /** Predicate that checks if a numeric value is not evenly divisible by 2. */
+  type Odd = Not[Even]
```

Comment:
This is very nice. I'm still delighted to see how we can build complex predicates just by composition. This is function composition at the type-level. :-)